src/Classes/Entities/EventEntity.php

```diff
@@ -28,5 +28,173 @@ public function __construct(
         $this->startTime = ($this->startTime ?? $startTime);
         $this->endTime = ($this->endTime ?? $endTime);
         $this->notes = ($this->notes ?? $notes);
+
+        $this->sanitiseData();
+    }
+
+    /**
+     * Will sanitise all the fields for an event.
+     */
+    private function sanitiseData()
+    {
+        $this->name = $this->sanitiseString($this->name);
+        $this->name = self::validateLength($this->name, 255);
+        $this->category = (int)$this->category;
+        $this->location = $this->sanitiseString($this->location);
+        $this->location = self::validateLength($this->location, 255);
+        $this->date = $this->sanitiseDate($this->date);
+        $this->startTime = $this->sanitiseTime($this->startTime);
+        $this->endTime = $this->sanitiseTime($this->endTime);
+        $this->notes = $this->sanitiseString($this->notes);
+        $this->notes = self::validateLength($this->notes, 255);
+    }
+
+    /**
+     * Sanitise as a date in the event table as data.
+     *
+     * @param $date
+     *
+     * @return bool|string
+     */
+    public function sanitiseDate($date)
```

Comment:
This and the time functions I would say are validating rather than sanitising - sanitisation normally means 'making the same data cleaner' while these are checking that the data matches an expected pattern.

Reply:
ok